types/ember/test/ember-module-tests.ts

```diff
@@ -0,0 +1,280 @@
+// setOwner
+Ember.setOwner(o2.create(), {});
+// setProperties
+Ember.setProperties(o2.create(), { name: 'bar' }).name; // $ExpectType string
+// tryInvoke
+Ember.tryInvoke(o2, 'init');
+Ember.tryInvoke(o2, 'init', [441]);
+// trySet
+Ember.trySet(o2, 'nam', ''); // $ExpectType any
+// typeOf
+Ember.typeOf(''); // $ExpectType "string"
+Ember.typeOf(Ember.A()); // $ExpectType "array"
+// warn
+Ember.warn('be caseful!');
+// VERSION
+Ember.VERSION; // $ExpectType string
+
+// onerror
+
+Ember.onerror = (err: Error) => console.error(err);
+Ember.onerror = (num: number, err: Error) => console.error(err); // $ExpectError
+
+// Classes
+// TODO ContainerProxyMixin
+// Ember
+// Ember.Application
+new Ember.Application(); // $ExpectType Application
+Ember.Application.create(); // $ExpectType Application
+// Ember.ApplicationInstance
+new Ember.ApplicationInstance(); // $ExpectType ApplicationInstance
+Ember.ApplicationInstance.create(); // $ExpectType ApplicationInstance
+// TODO: Ember.ApplicationInstance.BootOptions
+// Ember.Array
+const a1: Ember.Array<string> = [];
+// Ember.ArrayProxy
+new Ember.ArrayProxy<number>([3, 3, 2]); // $ExpectType ArrayProxy<number>
+// Ember.Checkbox
+const cb = new Ember.Checkbox(); // $ExpectType Checkbox
+cb.tagName; // $ExpectType string
+// Ember.Component
+const C1 = Ember.Component.extend({ classNames: ['foo'] });
+class C2 extends Ember.Component {
+    classNames = ['foo'];
+}
+const c1 = new C1();
+const c2 = new C2();
+C1.create();
+C2.create();
+c1.didInsertElement();
+c2.didInsertElement();
+// Ember.ComputedProperty
+const cp: Ember.ComputedProperty<string, string> = Ember.computed('foo', {
+    get(): string {
+        return '';
+    },
+    set(_key: string, newVal: string): string {
+        return '';
+    }
+});
+// Ember.ContainerDebugAdapter
+const cda = new Ember.ContainerDebugAdapter(); // $ExpectType ContainerDebugAdapter
+// Ember.Controller
+const con = new Ember.Controller(); // $ExpectType Controller
+// Ember.CoreObject
+const co = new Ember.CoreObject(); // $ExpectType CoreObject
+// Ember.DataAdapter
+const da = new Ember.DataAdapter(); // $ExpectType DataAdapter
+// Ember.Debug
+Ember.Debug.registerDeprecationHandler(() => {});
+Ember.Debug.registerWarnHandler(() => {});
+// Ember.DefaultResolver
+const dr = new Ember.DefaultResolver();
+dr.resolve('route:index');
+// Ember.Engine
+const e1 = new Ember.Engine();
+e1.register('data:foo', {}, { instantiate: false });
+// Ember.EngineInstance
+const ei1 = new Ember.EngineInstance();
+ei1.lookup('data:foo');
+// Ember.Error
+new Ember.Error('Halp!');
+// Ember.Evented
+const oe1 = EmberObject.extend(Evented).create();
+oe1.trigger('foo');
+oe1.on('bar', () => {});
+// Ember.HashLocation
+const hl = new Ember.HashLocation();
+// Ember.Helper
+const h1 = Ember.Helper.extend({
+    compute() {
+        this.recompute();
+        return '';
+    }
+});
+// Ember.HistoryLocation
+const hil = new Ember.HistoryLocation();
```

Comment:
loads of history location methods that aren't tested here

Reply:
These are not typed yet, so I can't write tests for them

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d9b2a03f896235ef0b4207ba313f0b74314227f6/types/ember/index.d.ts#L887-L897